File: GrooveOn.Models/ResponseObjects/PlaylistSongResponse.cs
using System;

namespace GrooveOn.Model.ResponseObjects
{
    public class PlaylistSongResponse
    {
        public int Id { get; set; }

        public int PlaylistId { get; set; }

        public string? PlaylistName { get; set; }

        public int SongId { get; set; }

        public string? SongTitle { get; set; }

        public string? ArtistName { get; set; }

        public string? CoverUrl { get; set; }

        public string? ExternalTrackId { get; set; }

        public int? DurationSeconds { get; set; }

        public DateTime AddedAt { get; set; }
    }
}

File: UI/grooveon_mobile/lib/models/playlist_song_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'playlist_song_response.g.dart';

@JsonSerializable()
class PlaylistSongResponse {
  final int id;

  final int playlistId;
  final String? playlistName;

  final int songId;
  final String? songTitle;

  final String? artistName;
  final String? coverUrl;

  final String? externalTrackId;
  final int? durationSeconds;

  final DateTime addedAt;

  PlaylistSongResponse({
    required this.id,
    required this.playlistId,
    this.playlistName,
    required this.songId,
    this.songTitle,
    this.artistName,
    this.coverUrl,
    this.externalTrackId,
    this.durationSeconds,
    required this.addedAt,
  });

  factory PlaylistSongResponse.fromJson(Map<String, dynamic> json) =>
      _$PlaylistSongResponseFromJson(json);

  Map<String, dynamic> toJson() => _$PlaylistSongResponseToJson(this);
}